src/ORM/FieldType/DBDate.php

```diff
@@ -432,85 +501,129 @@ public function InFuture()
+    /**
+     * Fix non-iso dates
+     *
+     * @param string $value
+     * @return string
+     */
+    protected function fixInputDate($value)
     {
- // 0 is a Monday
-        return (((mktime(0, 0, 0, $fmonth, $fday, $fyear) - mktime(0, 0, 0, 7, 17, 2006))/(60*60*24))+700000) % 7;
-    }
+        // split
+        list($day, $month, $year, $time) = $this->explodeDateString($value);
 
-    public function prior_monday($fyear, $fmonth, $fday)
-    {
-        return date("Y-m-d", mktime(0, 0, 0, $fmonth, $fday-$this->weekday($fyear, $fmonth, $fday), $fyear));
+        // Detect invalid year order
+        if (!checkdate($month, $day, $year) && checkdate($month, $year, $day)) {
+            trigger_error(
+                "Unexpected date order. Use " . self::ISO_DATE . " to prevent this notice.",
+                E_USER_NOTICE
+            );
+            list($day, $year) = [$year, $day];
+        }
+
+        // Fix y2k year
+        $year = $this->guessY2kYear($year);
+
+        // Validate date
+        if (!checkdate($month, $day, $year)) {
+            throw new InvalidArgumentException("Invalid date passed. Use " . self::ISO_DATE . " to prevent this error.");
+        }
+
+        // Convert to y-m-d
+        return sprintf('%d-%02d-%02d%s', $year, $month, $day, $time);
     }
 
     /**
-     * Return the nearest date in the past, based on day and month.
-     * Automatically attaches the correct year.
-     *
-     * This is useful for determining a financial year start or end date.
+     * Attempt to split date string into day, month, year, and timestamp components.
+     * Don't read this code without a drink in hand!
      *
-     * @param $fmonth int The number of the month (e.g. 3 is March, 4 is April)
-     * @param $fday int The day of the month
-     * @param $fyear int Determine historical value
-     * @return string Date in YYYY-MM-DD format
+     * @param string $value
+     * @return array
      */
-    public static function past_date($fmonth, $fday = 1, $fyear = null)
+    protected function explodeDateString($value)
     {
-        if (!$fyear) {
-            $fyear = date('Y');
+        // US date format with 4-digit year first
+        if (preg_match('#^(?<year>\\d{4})/(?<day>\\d+)/(?<month>\\d+)(?<time>.*)$#', $value, $matches)) {
+            trigger_error(
+                "Implicit y/d/m conversion. Use " . self::ISO_DATE . " to prevent this notice.",
+                E_USER_NOTICE
+            );
+            return [$matches['day'], $matches['month'], $matches['year'], $matches['time']];
         }
-        $fday = (int) $fday;
-        $fmonth = (int) $fmonth;
-        $fyear = (int) $fyear;
 
-        $pastDate = mktime(0, 0, 0, $fmonth, $fday, $fyear);
-        $curDate = mktime(0, 0, 0, date('m'), date('d'), $fyear);
+        // US date format without 4-digit year first: assume m/d/y
+        if (preg_match('#^(?<month>\\d+)/(?<day>\\d+)/(?<year>\\d+)(?<time>.*)$#', $value, $matches)) {
```

Comment:
How do you know that it's `m/d/y` here? Could be NZ date format with a short year (`d/m/y`), e.g. `31/07/17`. I'd change the error message to "Ambiguous short year format detected. Use <ISO_DATE> to prevent this notice". Also, add the value to the error message.